src/lib/github-enterprise/github-enterprise-contributors.ts

```diff
@@ -0,0 +1,185 @@
+import {
+  GithubEnterpriseTarget,
+  Username,
+  Contributor,
+  ContributorMap,
+} from '../types';
+import { Commits, Repo, Org } from './types';
+import { fetchAllPages } from './utils';
+
+import * as debugLib from 'debug';
+const debug = debugLib('snyk:github-enterprise-count');
+
+export const fetchGithubEnterpriseContributors = async (
+  githubEnterpriseInfo: GithubEnterpriseTarget,
+  threeMonthsDate: string,
+): Promise<ContributorMap> => {
+  const contributorsMap = new Map<Username, Contributor>();
+  try {
+    let repoList: Repo[] = [];
+    let orgList: Org[] = [];
+    if (
+      githubEnterpriseInfo.repo &&
+      (!githubEnterpriseInfo.orgs || githubEnterpriseInfo.orgs.length > 1)
+    ) {
+      // If repo is specified, then a single project key is expected, bail otherwise
+      console.log(
+        'You must provide a single org name for single repo counting',
+      );
+      process.exit(1);
+    } else if (githubEnterpriseInfo.repo && githubEnterpriseInfo.orgs) {
+      // If repo is specified, and we got a single project key
+      debug('Counting contributors for single repo');
+      repoList.push({
+        name: githubEnterpriseInfo.repo,
+        owner: { login: githubEnterpriseInfo.orgs[0] },
+      });
+    } else if (!githubEnterpriseInfo.orgs) {
+      githubEnterpriseInfo.orgs = [];
+      orgList = orgList.concat(
+        await fetchGithubEnterpriseOrgs(
+          githubEnterpriseInfo.fetchAllOrgs
+            ? `${githubEnterpriseInfo.url}api/v3/organizations?per_page=100`
+            : `${githubEnterpriseInfo.url}api/v3/user/orgs?per_page=100`,
+          githubEnterpriseInfo.token,
+        ),
+      );
+      for (let i = 0; i < orgList.length; i++) {
+        githubEnterpriseInfo.orgs.push(orgList[i].login);
+      }
+      repoList = repoList.concat(
```

Comment:
Take this out of the if else since it's the same in both cases, no?

Reply:
It breaks the script. It's similar but not exactly the same since if we provide a specific repo, it will still go and get all the repos..
I can add if(no specific org was given) but it's the same result. leaving it be for now.